packages/client/src/utils/persistence/persistenceMiddleware.ts

```diff
@@ -0,0 +1,105 @@
+import { USER_DETAILS_AVAILABLE } from '@client/profile/profileActions'
+import { IStoreState } from '@client/store'
+import {
+  CORRECTION_TOTALS,
+  PERFORMANCE_METRICS,
+  PERFORMANCE_STATS
+} from '@client/views/SysAdmin/Performance/metricsQuery'
+import { Action, Middleware } from 'redux'
+import { getDefaultPerformanceLocationId } from '@client/navigation'
+import { IUserDetails } from '@client/utils/userUtils'
+import {
+  FIELD_AGENT_ROLES,
+  NATIONAL_REGISTRAR_ROLES,
+  NATL_ADMIN_ROLES
+} from '@client/utils/constants'
+import { client } from '@client/utils/apolloClient'
+import { READY } from '@client/offline/actions'
+import startOfMonth from 'date-fns/startOfMonth'
+import subMonths from 'date-fns/subMonths'
+
+const isUserOfNationalScope = (userDetails: IUserDetails) =>
+  [...NATIONAL_REGISTRAR_ROLES, ...NATL_ADMIN_ROLES].includes(userDetails.role!)
+const isFieldAgent = (userDetails: IUserDetails) =>
+  FIELD_AGENT_ROLES.includes(userDetails?.role as string)
+
+function getQueriesToPrefetch(locationId: string, officeSelected: boolean) {
+  const defaultTimeStart = new Date(
+    startOfMonth(subMonths(new Date(Date.now()), 11)).setHours(0, 0, 0, 0)
+  )
+  const defaultTimeEnd = new Date(new Date(Date.now()).setHours(0, 0, 0, 0))
+
+  return [
+    {
+      query: PERFORMANCE_STATS,
+      variables: {
+        locationId,
+        status: [
+          'IN_PROGRESS',
+          'DECLARED',
+          'REJECTED',
+          'VALIDATED',
+          'WAITING_VALIDATION',
+          'REGISTERED'
+        ],
+        populationYear: new Date().getFullYear(),
+        officeSelected
+      },
+      fetchPolicy: 'cache-first'
```

Comment:
How about in the middleware we use `cache-and-network`, and `cache-first` in the views. That way when the app loads, we update the cache with new data if any and the views only query the cache

Reply:
Agreed, will update accordingly, thanks!